src/Planners/src/SwingFootPlanner.cpp

```diff
@@ -0,0 +1,269 @@
+    }
+
+    if (!handler->getParameter("step_height", m_stepHeight))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the step height for "
+                     "the SwingFootPlanner.";
+        return false;
+    }
+
+    if (!handler->getParameter("foot_apex_time", m_footApexTime))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the foot apex time for "
+                     "the SwingFootPlanner.";
+        return false;
+    }
+
+    double footLandingVelocity;
+    if (!handler->getParameter("foot_landing_velocity", footLandingVelocity))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the foot landing "
+                     "velocity for the SwingFootPlanner.";
+        return false;
+    }
+
+    double footLandingAcceleration;
+    if (!handler->getParameter("foot_landing_acceleration", footLandingAcceleration))
+    {
+        std::cerr << "[SwingFootPlanner::initialize] Unable to initialize the foot landing "
+                     "acceleration for the SwingFootPlanner.";
+        return false;
+    }
+
+    // check the parameters passed to the planner
+    bool ok = (m_dT > 0) && ((m_footApexTime > 0) && (m_footApexTime < 1));
+    if (!ok)
+    {
+        std::cerr << "[SwingFootPlanner::initialize] There is a problem in the parameters used to "
+                     "configure the planner. Please remember to set a strictly positive "
+                     "sampling_time, and a foot_apex_time strictly between 0 and 1."
+                  << std::endl;
+        return false;
+    }
+
+    m_planarPlanner.setInitialConditions(Eigen::Vector2d::Zero(), Eigen::Vector2d::Zero());
+    m_planarPlanner.setFinalConditions(Eigen::Vector2d::Zero(), Eigen::Vector2d::Zero());
+
+
+    m_heightPlanner.setInitialConditions(Vector1d::Zero(), Vector1d::Zero());
+    m_heightPlanner.setFinalConditions(Vector1d::Constant(footLandingVelocity), Vector1d::Constant(footLandingAcceleration));
+
+    return true;
+}
+
+void SwingFootPlanner::setContactList(const ContactList& contactList)
+{
+    // reset the time
+    m_currentTrajectoryTime = 0;
+
+    m_contactList = contactList;
+
+    // set the first contact
+    m_currentContactPtr = m_contactList.firstContact();
+    m_state.transform = m_currentContactPtr->pose;
+    m_state.spatialVelocity.setZero();
+    m_state.spatialAcceleration.setZero();
+    m_state.isInContact = true;
+
+}
+
+bool SwingFootPlanner::isValid() const
+{
+    return m_contactList.size() != 0;
+}
+
+const SwingFootPlannerState& SwingFootPlanner::get() const
+{
+    return m_state;
+}
+
+bool SwingFootPlanner::updateSE3Traj()
+{
+    // compute the trajectory at the current time
+    const double shiftedTime = m_currentTrajectoryTime - m_currentContactPtr->deactivationTime;
+
+    manif::SO3d rotation;
+    auto angularVelocity(m_state.spatialVelocity.asSO3());
+    auto angularAcceleration(m_state.spatialAcceleration.asSO3());
```

Comment:
Here, we are lucky because spatial angular velocity and mixed trivialized angular velocity are the same. However, when dealing with linear components, the naming might become problematic.

Reply:
I added a comment here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/90/commits/b81089be999fcf85e6495715fad1dd7fce83c56c

Reply:
Actually, here the `manif` method `asSO3()` for a twist is a bit confusing since it might mean that the twist is lifted to the SO(3) space through an exponential map after a hat operation. Could you add a comment as to say that we are only accessing the angular part of the twist using this method?

Reply:
Done https://github.com/dic-iit/bipedal-locomotion-framework/pull/90/commits/1560108a3ec368b5aa9ebc3ef1cdea936456d6c3